pkg/registration/registration_client.go

```diff
@@ -87,32 +88,72 @@ func (rClient *K8sRegistrationClient) GetActionPolicy() []*proto.ActionPolicyDTO
 	podPolicy[proto.ActionItemDTO_MOVE] = supported
 	podPolicy[proto.ActionItemDTO_PROVISION] = supported
 	podPolicy[proto.ActionItemDTO_RIGHT_SIZE] = notSupported
-	addActionPolicy(ab, pod, podPolicy)
+	rClient.addActionPolicy(ab, pod, podPolicy)
 
 	//2. container: support resize; recommend provision; not move;
 	container := proto.EntityDTO_CONTAINER
 	containerPolicy := make(map[proto.ActionItemDTO_ActionType]proto.ActionPolicyDTO_ActionCapability)
 	containerPolicy[proto.ActionItemDTO_RIGHT_SIZE] = supported
 	containerPolicy[proto.ActionItemDTO_PROVISION] = recommend
 	containerPolicy[proto.ActionItemDTO_MOVE] = notSupported
-	addActionPolicy(ab, container, containerPolicy)
+	rClient.addActionPolicy(ab, container, containerPolicy)
 
 	//3. application: only recommend provision; all else are not supported
 	app := proto.EntityDTO_APPLICATION
 	appPolicy := make(map[proto.ActionItemDTO_ActionType]proto.ActionPolicyDTO_ActionCapability)
 	appPolicy[proto.ActionItemDTO_PROVISION] = recommend
 	appPolicy[proto.ActionItemDTO_RIGHT_SIZE] = notSupported
 	appPolicy[proto.ActionItemDTO_MOVE] = notSupported
-	addActionPolicy(ab, app, appPolicy)
+	rClient.addActionPolicy(ab, app, appPolicy)
 
 	return ab.Create()
 }
 
-func addActionPolicy(ab *builder.ActionPolicyBuilder,
+func (rClient *K8sRegistrationClient) addActionPolicy(ab *builder.ActionPolicyBuilder,
 	entity proto.EntityDTO_EntityType,
 	policies map[proto.ActionItemDTO_ActionType]proto.ActionPolicyDTO_ActionCapability) {
 
 	for action, policy := range policies {
 		ab.WithEntityActions(entity, action, policy)
 	}
 }
+
+func (rclient *K8sRegistrationClient) GetEntityMetadata() []*proto.EntityIdentityMetadata {
+	glog.V(3).Infof("Begin to build EntityIdentityMetadata")
+
+	result := []*proto.EntityIdentityMetadata{}
+
+	entities := []proto.EntityDTO_EntityType{
```

Comment:
How about VDC? We should also include it.

Reply:
yes, just added it.